### ISSUE TITLE
Cashbox Template Generation causes wrong formatting of mysql connectionstring

### ISSUE THREAD

**Issue by darek-phorest** [fiskaltrust/product-de-bring-your-own-datacenter#41]
**Describe the bug**

HI we are seeing intermittent issues with generating a cashbox and queue using our template:
here is our template:
```
{
  "ftCashBoxId": "|[cashbox_id]|",
  "ftSignaturCreationDevices": [
    {
      "Id": "|[scu0_id]|",
      "Package": "fiskaltrust.Middleware.SCU.DE.Fiskaly",
      "Configuration": {
        "ApiKey": "|[fyapikey]|",
        "ApiSecret": "|[fyapisecret]|",
        "TssId": "|[fytssid]|"
      },
      "Url": [
        "grpc://localhost:18004"
      ]
    }
  ],
  "ftQueues": [
    {
      "Id": "|[queue0_id]|",
      "Package": "fiskaltrust.Middleware.Queue.MySQL",
      "Configuration": {
        "connectionstring": "|[db_connection_string]|",
        "init_ftQueue": [
          {
            "ftQueueId": "|[queue0_id]|",
            "ftCashBoxId": "|[cashbox_id]|",
            "CountryCode": "DE",
            "Timeout": 15000
          }
        ],
        "init_ftQueueDE": [
          {
            "ftQueueDEId": "|[queue0_id]|",
            "ftSignaturCreationUnitDEId": "|[scu0_id]|",
            "CashBoxIdentification": "phorest|[queue0_id_base64withoutspecialchars]|"
          }
        ],
        "init_ftSignaturCreationUnitAT": [],
        "init_ftSignaturCreationUnitDE": [
          {
            "ftSignaturCreationUnitDEId": "|[scu0_id]|",
            "Url": "[\"grpc://localhost:18004\"]"
          }
        ],
        "init_ftSignaturCreationUnitFR": [],
        "init_ftCashBox": {
          "ftCashBoxId": "|[cashbox_id]|"
        }
      },
      "Url": [
        "rest://localhost:1500/|[queue0_id]|"
      ]
    }
  ]
}
```

and here is the error we are seeing in byodc: 

```
41db8dcc17ebd31ed30b' at line 1)\\n       ---> MySqlConnector.MySqlException (0x80004005): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '679e0018db4441db8dcc17ebd31ed30b; USE 679e0018db4441db8dcc17ebd31ed30b' at line 1\\n       ---> MySqlConnector.MySqlException (0x80004005): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '679e0018db4441db8dcc17ebd31ed30b; USE 679e0018db4441db8dcc17ebd31ed30b' at line 1\\n         at MySqlConnector.Core.ServerSession.ReceiveReplyAsyncAwaited(ValueTask`1 task) in //src/MySqlConnector/Core/ServerSession.cs:line 817\\n         at MySqlConnector.Core.ResultSet.ReadResultSetHeaderAsync(IOBehavior ioBehavior) in //src/MySqlConnector/Core/ResultSet.cs:line 49\\n         at MySqlConnector.MySqlDataReader.ActivateResultSet(CancellationToken cancellationToken) in //src/MySqlConnector/MySqlDataReader.cs:line 118\\n         at MySqlConnector.MySqlDataReader.CreateAsync(CommandListPosition commandListPosition, ICommandPayloadCreator payloadCreator, IDictionary`2 cachedProcedures, IMySqlCommand command, CommandBehavior behavior, IOBehavior ioBehavior, CancellationToken cancellationToken) in //src/MySqlConnector/MySqlDataReader.cs:line 436\\n         at MySqlConnector.Core.CommandExecutor.ExecuteReaderAsync(IReadOnlyList`1 commands, ICommandPayloadCreator payloadCreator, CommandBehavior behavior, IOBehavior ioBehavior, CancellationToken cancellationToken) in //src/MySqlConnector/Core/CommandExecutor.cs:line 60\\n         at MySqlConnector.MySqlCommand.ExecuteNonQueryAsync(IOBehavior ioBehavior, CancellationToken cancellationToken) in //src/MySqlConnector/MySqlCommand.cs:line 264\\n         at Dapper.SqlMapper.ExecuteImplAsync(IDbConnection cnn, CommandDefinition command, Object param) in /_/Dapper/SqlMapper.Async.cs:line 687\\n         at fiskaltrust.Middleware.Storage.MySQL.DatabaseInitialization.DatabaseMigrator.MigrateAsync()\\n         at fiskaltrust.Middleware.Storage.MySQL.MySQLBootstrapper.InitAsync(Guid queueId, Dictionary`2 configuration, ILogger`1 logger)\\n         --- End of inner exception stack trace ---\\n         at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)\\n         at fiskaltrust.Middleware.Storage.MySQL.MySQLBootstrapper.ConfigureStorageServices(IServiceCollection serviceCollection)\\n         at fiskaltrust.Middleware.Queue.MySQL.PosBootstrapper.ConfigureServices(IServiceCollection serviceCollection)\\n         at fiskaltrust.SignatureCloud.DE.Services.MiddlewareInstanceFactory.CreateQueue(ftCashBoxConfiguration configuration) in /src/src/fiskaltrust.SignatureCloud.DE/Services/MiddlewareInstanceFactory.cs:line 211\\n         at fiskaltrust.SignatureCloud.DE.Services.MiddlewareInstanceFactory.<>c__DisplayClass10_0.<<CreateAsync>b__0>d.MoveNext() in /src/src/fiskaltrust.SignatureCloud.DE/Services/MiddlewareInstanceFactory.cs:line 185\\n      --- End of stack trace from previous location ---\\n         at fiskaltrust.SignatureCloud.DE.Services.Locking.RedisLockService.ProcessLockiningAsync[T](String lockName, Func`1 executeMethod) in /src/src/fiskaltrust.SignatureCloud.DE/Services/Locking/RedisLockService.cs:line 36\\n         at
2:10
fiskaltrust.SignatureCloud.DE.Services.MiddlewareInstanceFactory.CreateAsync(Guid cashboxid, String accesstoken, ftCashBoxConfiguration configuration) in /src/src/fiskaltrust.SignatureCloud.DE/Services/MiddlewareInstanceFactory.cs:line 187\\n         at fiskaltrust.SignatureCloud.DE.Controllers.SignController.Post(ReceiptRequest request) in /src/src/fiskaltrust.SignatureCloud.DE/Controllers/SignController.cs:line 102\\n         at lambda_method11(Closure , Object )\\n         at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.AwaitableObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)\\n         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)\\n         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)\\n         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)\\n         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)\\n         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)\\n         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)\\n         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)\\n         at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)\\n         at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)\\n         at Microsoft.AspNetCore.Builder.Extensions.MapWhenMiddleware.Invoke(HttpContext context)\\n         at Microsoft.AspNetCore.Builder.Extensions.MapWhenMiddleware.Invoke(HttpContext context)\\n         at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)\\n\",\"stream\":\"stdout\",\"docker\":{\"container_id\":\"d94347dfe48e156f5760acb6eaf47141dfc58ac49df8102798c81f319d327bd4\"},\"kubernetes\":{\"container_name\":\"byodc\",\"namespace_name\":\"bring-your-own-datacenter\",\"pod_name\":\"byodc-5d6579fbff-zbgl9\",\"container_image\":\"ghcr.io/fiskaltrust/byodc-mysql-fiskaly:1.3.16-buster\",\"container_image_id\":\"docker-pullable://ghcr.io/fiskaltrust/byodc-mysql-fiskaly@sha256:7443476fc54d0f2233758b9bd499ddf660b7fe35d27cb3d049c33b565cc6f271\",\"pod_id\":\"683adb66-ebf8-452e-b2c6-a57260f7680b\",\"host\":\"ip-192-168-128-81.eu-central-1.compute.internal\",\"labels\":{\"app\":\"byodc\",\"pod-template-hash\":\"5d6579fbff\"},\"master_url\":\"https://10.100.0.1:443/api\",\"namespace_id\":\"55e764f3-dcc1-456b-a56a-02b4edd83cd7\"}}"}
```

in the above it looks like the connection string generated is not properly formatted.
we need to generate a new cashbox with the same template and the same parameters and usually the second run that creates a new cashbox and queue resolves the issue which is strange because it is the same template and the same template variables, nothing changes on our side in that regard

**To Reproduce**

**Expected behavior**
the cashbox and queue are properly generated in fiskaltrust portal

**Screenshots**

**STDOUT/STDERR**
```
41db8dcc17ebd31ed30b' at line 1)\\n       ---> MySqlConnector.MySqlException (0x80004005): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '679e0018db4441db8dcc17ebd31ed30b; USE 679e0018db4441db8dcc17ebd31ed30b' at line 1\\n       ---> MySqlConnector.MySqlException (0x80004005): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '679e0018db4441db8dcc17ebd31ed30b; USE 679e0018db4441db8dcc17ebd31ed30b' at line 1\\n         at MySqlConnector.Core.ServerSession.ReceiveReplyAsyncAwaited(ValueTask`1 task) in //src/MySqlConnector/Core/ServerSession.cs:line 817\\n         at MySqlConnector.Core.ResultSet.ReadResultSetHeaderAsync(IOBehavior ioBehavior) in //src/MySqlConnector/Core/ResultSet.cs:line 49\\n         at MySqlConnector.MySqlDataReader.ActivateResultSet(CancellationToken cancellationToken) in //src/MySqlConnector/MySqlDataReader.cs:line 118\\n         at MySqlConnector.MySqlDataReader.CreateAsync(CommandListPosition commandListPosition, ICommandPayloadCreator payloadCreator, IDictionary`2 cachedProcedures, IMySqlCommand command, CommandBehavior behavior, IOBehavior ioBehavior, CancellationToken cancellationToken) in //src/MySqlConnector/MySqlDataReader.cs:line 436\\n         at MySqlConnector.Core.CommandExecutor.ExecuteReaderAsync(IReadOnlyList`1 commands, ICommandPayloadCreator payloadCreator, CommandBehavior behavior, IOBehavior ioBehavior, CancellationToken cancellationToken) in //src/MySqlConnector/Core/CommandExecutor.cs:line 60\\n         at MySqlConnector.MySqlCommand.ExecuteNonQueryAsync(IOBehavior ioBehavior, CancellationToken cancellationToken) in //src/MySqlConnector/MySqlCommand.cs:line 264\\n         at Dapper.SqlMapper.ExecuteImplAsync(IDbConnection cnn, CommandDefinition command, Object param) in /_/Dapper/SqlMapper.Async.cs:line 687\\n         at fiskaltrust.Middleware.Storage.MySQL.DatabaseInitialization.DatabaseMigrator.MigrateAsync()\\n         at fiskaltrust.Middleware.Storage.MySQL.MySQLBootstrapper.InitAsync(Guid queueId, Dictionary`2 configuration, ILogger`1 logger)\\n         --- End of inner exception stack trace ---\\n         at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)\\n         at fiskaltrust.Middleware.Storage.MySQL.MySQLBootstrapper.ConfigureStorageServices(IServiceCollection serviceCollection)\\n         at fiskaltrust.Middleware.Queue.MySQL.PosBootstrapper.ConfigureServices(IServiceCollection serviceCollection)\\n         at fiskaltrust.SignatureCloud.DE.Services.MiddlewareInstanceFactory.CreateQueue(ftCashBoxConfiguration configuration) in /src/src/fiskaltrust.SignatureCloud.DE/Services/MiddlewareInstanceFactory.cs:line 211\\n         at fiskaltrust.SignatureCloud.DE.Services.MiddlewareInstanceFactory.<>c__DisplayClass10_0.<<CreateAsync>b__0>d.MoveNext() in /src/src/fiskaltrust.SignatureCloud.DE/Services/MiddlewareInstanceFactory.cs:line 185\\n      --- End of stack trace from previous location ---\\n         at fiskaltrust.SignatureCloud.DE.Services.Locking.RedisLockService.ProcessLockiningAsync[T](String lockName, Func`1 executeMethod) in /src/src/fiskaltrust.SignatureCloud.DE/Services/Locking/RedisLockService.cs:line 36\\n         at
2:10
fiskaltrust.SignatureCloud.DE.Services.MiddlewareInstanceFactory.CreateAsync(Guid cashboxid, String accesstoken, ftCashBoxConfiguration configuration) in /src/src/fiskaltrust.SignatureCloud.DE/Services/MiddlewareInstanceFactory.cs:line 187\\n         at fiskaltrust.SignatureCloud.DE.Controllers.SignController.Post(ReceiptRequest request) in /src/src/fiskaltrust.SignatureCloud.DE/Controllers/SignController.cs:line 102\\n         at lambda_method11(Closure , Object )\\n         at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.AwaitableObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)\\n         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)\\n         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)\\n         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)\\n         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)\\n         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)\\n         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)\\n         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)\\n         at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)\\n         at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)\\n         at Microsoft.AspNetCore.Builder.Extensions.MapWhenMiddleware.Invoke(HttpContext context)\\n         at Microsoft.AspNetCore.Builder.Extensions.MapWhenMiddleware.Invoke(HttpContext context)\\n         at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)\\n\",\"stream\":\"stdout\",\"docker\":{\"container_id\":\"d94347dfe48e156f5760acb6eaf47141dfc58ac49df8102798c81f319d327bd4\"},\"kubernetes\":{\"container_name\":\"byodc\",\"namespace_name\":\"bring-your-own-datacenter\",\"pod_name\":\"byodc-5d6579fbff-zbgl9\",\"container_image\":\"ghcr.io/fiskaltrust/byodc-mysql-fiskaly:1.3.16-buster\",\"container_image_id\":\"docker-pullable://ghcr.io/fiskaltrust/byodc-mysql-fiskaly@sha256:7443476fc54d0f2233758b9bd499ddf660b7fe35d27cb3d049c33b565cc6f271\",\"pod_id\":\"683adb66-ebf8-452e-b2c6-a57260f7680b\",\"host\":\"ip-192-168-128-81.eu-central-1.compute.internal\",\"labels\":{\"app\":\"byodc\",\"pod-template-hash\":\"5d6579fbff\"},\"master_url\":\"https://10.100.0.1:443/api\",\"namespace_id\":\"55e764f3-dcc1-456b-a56a-02b4edd83cd7\"}}"}
```
**POSSystem (please complete the following information):**
46a60bc4-42f6-4a95-b758-77c8da74f203

**Cashbox Information (please complete the following information):**
 - CashboxID

**Additional context**

**Comment by darek-phorest** [comment]
hi there, wanted to ask if there is anyway to decrypt the mysql connections string that comes back from the api together with the config: `eVCirX3dFUa1IR6xLhty8dNFX/6lO7JXSKGlOlJYOuga88lNBckQ4dqvKesUCZzv135mfQvM3udpUYqijcedsAiglNb2shnuCM1Kjl3MvkZ6ayzSvGseO1rFghuXe4nHHKktuJQbAMSziRL5YPAJ0S8UPZ1oF7mnpdHyicpEsSo=`

we want to verify that this is somehow configured in a wrong way when we use the template and receipts are failing because of the above reason

**Comment by Solrosy** [comment]
Could we please get an update on this?

**Comment by christian-rogobete** [comment]
Hi @darek-phorest, thank you for reporting this issue. It seams that the passed connection string by query parameter is not correct. Unfortunately there is no way to get the connection string from the base64 string because it is encrypted. Did you try to place the connection string directly into the template and not pass it by query parameter. If doing so is the issue still there?

**Comment by darek-phorest** [comment]
hi @christian-rogobete 

so we are trying to debug what exactly is happening, i am reluctant to say that the issue is on our side because if we run the same request with the same query parameters (so no changes on our side) then the cashbox is configured properly. So it fails on some occasions and we run the template again and it's fine.  

So the error : `MySQL server version for the right syntax to use near '679e0018db4441db8dcc17ebd31ed30b; USE 679e0018db4441db8dcc17ebd31ed30b'` would indicate that the sql somehow is not enclosed properly with quotes but that's just my assumption, we don't pass the `679e0018db4441db8dcc17ebd31ed30b` as it's being generated on your side, so the error would indicate that certain queue ids might produce this error but that's something you would need to verify on your side.

Let me know if i can provide you with anything that would help to investigate this

**Comment by christian-rogobete** [comment]
hi @darek-phorest
thank you for message and thanks again for reporting this issue. We have probably found the problem in our implementation and it will be fixed the upcoming release 1.3.19. The issue occurs for certain queue ids that start with a digit.

**Comment by darek-phorest** [comment]
hi @christian-rogobete 

great you were able to find what the issue is, can you let us know when 1.3.19 is released ?

**Comment by StefanKert** [comment]
@darek-phorest yes we have been able to find the issue. We plan to release the Middleware until the end of this week so the upgrades for the ByOD helm templates should follow next week.

**Comment by TSchmiedlechner** [comment]
Closing this now as 1.3.19 was released a few weeks ago. If the issue still persists, please feel free to get back to us anytime. Thank you very much for reporting this!

**Comment by darek-phorest** [comment]
Hi @TSchmiedlechner was the byodc released also?, we can only see the 1.3.18 in the charts: https://charts.fiskaltrust.cloud/

**Comment by darek-phorest** [comment]
Hi @TSchmiedlechner any update on 1.3.19 getting into charts: https://charts.fiskaltrust.cloud/ ?

**Comment by TSchmiedlechner** [comment]
Hi @darek-phorest, indeed it looks like this wasn't released properly, apologies for this. We will find out the cause and release the missing versions on Monday, and get back to you here when it's done.

**Comment by TSchmiedlechner** [comment]
Hi @darek-phorest, IMO this should be okay. We have not updated our charts, as they automatically use the `1.3-buster` tag of the image, which was released both for version 1.3.19 and 1.3.20: https://github.com/fiskaltrust/product-de-bring-your-own-datacenter/pkgs/container/byodc-mysql-fiskaly

You might need to update your deployment to the latest tag ([as described here at 2)](https://github.com/fiskaltrust/product-de-bring-your-own-datacenter/blob/master/OperationsReference.md#update-deployment))
